From f6269cadde64ac535092a74cb4fc493cc25014fe Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Sun, 1 Nov 2020 18:19:13 +0000
Subject: [PATCH 501/530] [libkworkspace] Fix if getCurrentSeat needs to
 fallback to old approach

The test for whether "/auto" is supported was bogus.

QDBusAbstractInterface::isValid effectively just confirms the service
exists and the path is a legally valid name, not that it has any
interfaces there.

This means the fallback path is not run appropriately.

BUG: 423526
---
 libkworkspace/kdisplaymanager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libkworkspace/kdisplaymanager.cpp
+++ b/libkworkspace/kdisplaymanager.cpp
@@ -431,7 +431,7 @@ static bool getCurrentSeat(QDBusObjectPa
     if (man.isValid()) {
         *currentSeat = QDBusObjectPath(_SYSTEMD_SEAT_BASE_PATH "/auto");
         SystemdSeat seat(*currentSeat);
-        if (seat.isValid()) {
+        if (seat.property("Id").isValid()) { //query an arbitrary property to confirm the path is valid
             return true;
         }
 
